modules/payloads/stages/php/meterpreter.rb

```diff
@@ -23,13 +24,22 @@ def initialize(info = {})
       'Session'       => Msf::Sessions::Meterpreter_Php_Php))
   end
 
-  def generate_stage
+  def generate_stage(opts={})
     file = File.join(Msf::Config.data_directory, "meterpreter", "meterpreter.php")
 
-    met = File.open(file, "rb") {|f|
+    met = File.open(file, "rb") { |f|
       f.read(f.stat.size)
     }
-    #met.gsub!(/#.*?$/, '')
+
+    uuid = opts[:uuid] || Msf::Payload::UUID.new(
+      platform: 'php',
+      arch:     ARCH_PHP
+    )
+
+    bytes = uuid.to_raw.chars.map { |c| '\x%.2x' % c.ord }.join('')
+    met = met.sub("\"PAYLOAD_UUID\", \"\"", "\"PAYLOAD_UUID\", \"#{bytes}\"")
+
+    met.gsub!(/#.*?$/, '')
     #met = Rex::Text.compress(met)
```

Comment:
Added space compression to the stage does ruin things, so it was not added here.